src/components/AutoUpdateTime.js

```diff
@@ -0,0 +1,102 @@
+import {View} from 'react-native';
+import React, {PureComponent} from 'react';
+import PropTypes from 'prop-types';
+import styles from '../styles/styles';
+import DateUtils from '../libs/DateUtils';
+import withLocalize, {withLocalizePropTypes} from './withLocalize';
+import Timers from '../libs/Timers';
+import Text from './Text';
+
+const propTypes = {
+    timezone: PropTypes.shape({
+        /** Value of selected timezone */
+        selected: PropTypes.string,
+
+        /** Whether timezone is automatically set */
+        automatic: PropTypes.bool,
+    }).isRequired,
+    ...withLocalizePropTypes,
+};
+
+class AutoUpdateTime extends PureComponent {
+    constructor(props) {
+        super(props);
+        this.getCurrentUserLocalTime = this.getCurrentUserLocalTime.bind(this);
+        this.runTimeInterval = this.runTimeInterval.bind(this);
+        this.updateCurrentTime = this.updateCurrentTime.bind(this);
+        this.state = {
+            timezone: this.getCurrentUserLocalTime(),
+        };
+    }
+
+    componentDidMount() {
+        this.updateCurrentTime();
+    }
+
+    componentDidUpdate() {
+        this.updateCurrentTime();
+    }
+
+    componentWillUnmount() {
+        clearInterval(this.timer);
+    }
+
+    getCurrentUserLocalTime() {
+        return DateUtils.getLocalMomentFromDatetime(
+            this.props.preferredLocale,
+            null,
+            this.props.timezone.selected,
+        );
+    }
+
+    /**
+     * Run the update time after {time} miliseconds.
+     * @param {*} time interval time in miliseconds.
+     */
+    runTimeInterval(time) {
+        if (this.timer) {
+            clearInterval(this.timer);
+            this.timer = null;
+        }
+        this.timer = Timers.register(
+            setInterval(() => {
+                this.setState({
+                    timezone: this.getCurrentUserLocalTime(),
+                });
+            }, time),
+        );
+    }
+
+    updateCurrentTime() {
+        const timeoutTime = (60 - this.state.timezone.seconds()) * 1000;
+        this.runTimeInterval(timeoutTime);
+    }
+
+    render() {
+        const GMTTime = this.state.timezone
+            ? `${this.state.timezone
+                .toString()
+                .split(/[+-]/)[0]
+                .slice(-3)} ${this.state.timezone.zoneAbbr()}`
+            : '';
+        const currentTime = this.state.timezone
+          && Number.isNaN(Number(this.state.timezone.zoneAbbr()))
+            ? this.state.timezone.zoneAbbr()
+            : GMTTime;
+        return (
+            <View style={[styles.mb6, styles.detailsPageSectionContainer]}>
+                <Text style={[styles.formLabel, styles.mb2]} numberOfLines={1}>
+                    {this.props.translate('detailsPage.localTime')}
+                </Text>
+                <Text numberOfLines={1}>
+                    {this.state.timezone.format('LT')}
+                    {' '}
+                    {currentTime}
```

Comment:
Can we get rid of most of this?
```suggestion
        return (
            <View style={[styles.mb6, styles.detailsPageSectionContainer]}>
                <Text style={[styles.formLabel, styles.mb2]} numberOfLines={1}>
                    {this.props.translate('detailsPage.localTime')}
                </Text>
                <Text numberOfLines={1}>
                    {this.state.timezone.format('LT')}
                    {' '}
                    {this.state.timezone.zoneAbbr()}
```

Reply:
@cead22 I think we still need to keep this, but we can remove condition related to this.state.timezone since it's required and can not be null. So this will be like:
```javascript
 const GMTTime = `${this.state.timezone
            .toString()
            .split(/[+-]/)[0]
            .slice(-3)} ${this.state.timezone.zoneAbbr()}`;
        const currentTime = Number.isNaN(Number(this.state.timezone.zoneAbbr()))
            ? this.state.timezone.zoneAbbr()
            : GMTTime;
```

Reply:
with GMT time it will be like: +06 +07, in this case we can't use `this.state.timezone.zoneAbbr()`.

Reply:
Is that condition backwards, should it be !Number.isNan? In any case, can you add a comment explaining it please?
```js
const currentTime = !Number.isNaN(Number(this.state.timezone.zoneAbbr()))
            ? this.state.timezone.zoneAbbr()
            : GMTTime;`
```

Reply:
Yeah sure, you can also test it by changing the timezone to Asia/Ho_Chi_Minh

Reply:
```javascript
        // With GMT timezone, the zoneAbbr() will return a number, in this case we should use the GMT ${number} format
        // ie: GMT +07
const currentTime = !Number.isNaN(Number(this.state.timezone.zoneAbbr()))
            ? this.state.timezone.zoneAbbr()
            : GMTTime;`
```
@cead22 are you ok with this comment?

Reply:
What does `this.state.timezone.zoneAbbr()` return for GMT and non-GMT, I still find that comment confusing, and it should also say **why**

Also, how do you feel about writing the code like this for clarity?
```js
            let currentTime;
            // This comment should explain what this case is for and why we check for isNaN
            if (Number.isNaN(Number(this.state.timezone.zoneAbbr())) {
                currentTime = this.state.timezone.zoneAbbr();
            } else {
                // This comment should explain how this.state.timezone looks like and what split/slice do
                currentTime = `${this.state.timezone
                    .toString()
                    .split(/[+-]/)[0]
                    .slice(-3)} ${this.state.timezone.zoneAbbr()}`;
            }
```